ui/src/components/modals/ModelListModal.js

```diff
@@ -0,0 +1,233 @@
+import React from 'react';
+import Component from 'react-component-component';
+import Spinner from 'react-spinkit';
+import moment from 'moment';
+import { get, omit } from 'lodash';
+
+import { api } from '@arranger/components';
+import globals from 'utils/globals';
+import tsvDownloader from 'utils/tsvDownloader';
+import modelImageProcessor from 'utils/modelImageProcessor';
+import { SelectedModelsContext } from 'providers/SelectedModels';
+import { ModalStateContext } from 'providers/ModalState';
+
+import { Row } from 'theme/system';
+import DoubleArrowRightIcon from 'icons/DoubleArrowRightIcon';
+import ModelPlaceholderIcon from 'icons/ModelPlaceholderIcon';
+import TrashIcon from 'icons/TrashIcon';
+import DownloadIconWhite from 'icons/DownloadIconWhite';
+import modelListEmptyRedPlus from 'assets/icon-modellist-empty-red.svg';
+
+const EmptyList = () => (
+  <div className="empty-list">
+    <img src={modelListEmptyRedPlus} alt="Add to list icon" width="26" height="30" />
+    <p>You don’t have any models selected.</p>
+    <a href="/">Browse models »</a>
+  </div>
+);
+
+const Loading = () => (
+  <Row justifyContent="center">
+    <Spinner
+      fadeIn="full"
+      name="ball-pulse-sync"
+      style={{
+        margin: 45,
+        width: 90,
+      }}
+    />
+  </Row>
+);
+
+const fetchData = async ({ setState, modelIds }) => {
```

Comment:
refactor this fetchData into a Query component like https://github.com/nci-hcmi-catalog/portal/blob/master/ui/src/components/Queries/AggregationQuery.js and reuse in the Model entity page?

Reply:
Yes I was thinking the exact same thing but Rosi was saying that she may need more fields but either way I think it would be good to have the queries pulled out of the components so I'll do that now and add an additional one if/when needed.